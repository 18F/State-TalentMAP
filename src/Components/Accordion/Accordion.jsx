import React from 'react';
import PropTypes from 'prop-types';

const shortid = require('shortid');

const Accordion = ({ items }) => (
  <ul className="usa-accordion">
    {
      items.map(item =>
        (
          <li key={shortid.generate()}>
            <button
              className="usa-accordion-button"
              aria-expanded="false"
              aria-controls={item.id}
            >
              {item.title}
            </button>
            <div id={item.id} className="usa-accordion-content">
              {item.content}
            </div>
          </li>
        ),
      )
    }
  </ul>
);

Accordion.propTypes = {
  items: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      title: PropTypes.string,
      content: PropTypes.node,
    }),
  ).isRequired,
};

export default Accordion;
