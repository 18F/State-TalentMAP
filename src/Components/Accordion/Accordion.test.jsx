import { shallow } from 'enzyme';
import React from 'react';
import toJSON from 'enzyme-to-json';
import Accordion from './Accordion';

describe('AccordionItemComponent', () => {
  const child = (
    <li>
      <button />
      <div className="usa-accordion-content" aria-hidden="false">
        <span />
      </div>
    </li>
  );

  it('can receive children element', () => {
    const wrapper = shallow(
      <Accordion>
        {child}
      </Accordion>,
    );
    expect(wrapper.instance().props.children).toBe(child);
  });

  it('can receive className prop', () => {
    const wrapper = shallow(
      <Accordion className="my-class">
        {child}
      </Accordion>,
    );
    expect(wrapper.find('.my-class').exists()).toBe(true);
  });

  it('can receive isMultiselectable prop', () => {
    const wrapper = shallow(
      <Accordion isMultiselectable>
        {child}
      </Accordion>,
    );
    expect(wrapper.find('ul').prop('aria-multiselectable')).toBe(true);
  });

  it('matches snapshot', () => {
    const wrapper = shallow(
      <Accordion>
        {child}
      </Accordion>,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
