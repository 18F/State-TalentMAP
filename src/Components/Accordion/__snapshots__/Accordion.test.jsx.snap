// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionItemComponent matches snapshot 1`] = `
<ul
  className="usa-accordion"
>
  <li>
    <button />
    <div
      aria-hidden="false"
      className="usa-accordion-content"
    >
      <span />
    </div>
  </li>
</ul>
`;
